src/PrestaShopBundle/Controller/Admin/Sell/Catalog/Product/AttachmentController.php

```diff
@@ -0,0 +1,63 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+declare(strict_types=1);
+
+namespace PrestaShopBundle\Controller\Admin\Sell\Catalog\Product;
```

Comment:
created this new controller here with the idea that controller reflects a page, so I didn't want to mix it all up in general AttachmentsController, as this one is responsible for product page related attachments stuff. Let me know what you think :up:

Reply:
Sice it's related to the association Product<->Attachment I think it's perfectly acceptable to create this controller
It keeps both the Product and Attachment controller clean and separate

Although since it's quite a small controller for now, it would have been added in the Product controller maybe, but definitely not in the already existing Attachment controller

Reply:
On second thought as I commented for the `GetProductAttachmentInfo` command, this is not related to the product at all It's a "generic" query only related to Attachment So the query, controller and everything related should be in the `Attachment` namespace

And this method could actually be moved to the existing AttachmentController